### PR TITLE
"ZipListT", newtype wrapper for a zippy Applicative instance.

### PR DESCRIPTION
**Pull request by mstksg**
Not sure if this is within the scope or the vision of this library, but I've been using this separately in a few places.  It's small enough (basically three lines) that it's probably not *necessary* to have it as a part of the library, but just putting it here in case you think it might be useful to have it in a central place.

### CODE REVIEW

File: src/List/Transformer.hs [hunk 1]
```diff
@@ -562,3 +566,46 @@ data Step m a = Cons a (ListT m a) | Nil
...
     fmap k (Cons x l) = Cons (k x) (fmap k l)
+
+-- | Similar to 'ZipList' in /base/: a newtype wrapper over 'ListT' that
+-- overrides its normal 'Applicative' instance (combine every combination)
+-- with one that "zips" outputs together one at a time.
+--
+-- >>> let xs = do x <- select [1,2,3,4]; liftIO (print x)
+-- >>> let ys = do y <- select [5,6]; liftIO (print y)
+-- >>> runListT (xs *> ys)
+-- "1"
```
Comment (Gabriella439) [inline src/List/Transformer.hs]: You will need to unquote the numbers according to CI

**Review by Gabriella439 — APPROVED** [review]
Yeah, this seems reasonable to me!

**Comment by mstksg** [comment]
CI seems to be passing as of 7500676 :)

**Comment by Gabriella439** [comment]
@mstksg: Thank you! 🙂

**Pull request merged**